pulsar/consumer_partition_test.go

```diff
@@ -37,6 +37,8 @@ func TestSingleMessageIDNoAckTracker(t *testing.T) {
 		metrics:              newTestMetrics(),
 		decryptor:            crypto.NewNoopDecryptor(),
 	}
+	pc.ackGroupingTracker = newAckGroupingTracker(&AckGroupingOptions{MaxSize: 0},
```

Comment:
Can you add more unit tests? Such as:
- MaxSize > 0
- A Individual ack in a  batch message.
- Other Similar to: [CPP test](https://github.com/apache/pulsar-client-cpp/blob/06eab69f4caedc1962a21ac52e5c19de868609cd/tests/AcknowledgeTest.cc)

Reply:
This line is to fix the error that `pc.ackGroupingTracker` is `nil` in this test without the explicit initialization of `pc.ackGroupingTracker` because the `partitionConsumer` is created manually, not by the `Consumer`. The tests for different `AckGroupingOptions` are added into `ack_grouping_tracker_test.go`.

> MaxSize > 0

The default config (`MaxSize: 1000, MaxTime: 10ms`) has been verified in all existing tests that call acknowledge related methods.

> A Individual ack in a batch message.

TestBatchIndexAck

This PR has already changed 600+ lines. Adding these tests could make it very huge. What do you think of adding the different options into a single test like `TestBatchIndexAck`?

Reply:
I've added the test cases to `TestBatchIndexAck` and fix the bug that `AckGroupingOptions` is not exposed, PTAL again.